### PR TITLE
Failed Transaction Fallback and Estimation of Gas for Registration

### LINKED ISSUE #231: Add button to start registration again
Nice to have: sometimes registrations get stucked :/ an easy way to explain a user to start over the process would be adding a button to do so.

### PR DESCRIPTION
**Pull request by jessgusclark**
# Exploring Gas Issues

Adds fallback during registration if the transaction fails at the miner level. Puts the user back to the previous step. Second, the estimation of gas cost for both registration transactions.

Rebased to Develop

Missing Translations

## Testing Instructions

These steps should be tested with both Nifty and Metamask on Chrome and Firefox. If testing on Local set the mining interval instead of automine.

1) request to register a domain
2) Change the gas limit to `26000`
3) **Expect** the transaction to fail and it shows a message, you are back at the beginning of step 1.
4) Don't change the gas limit and let it choose it for you
5) **Expect** transaction to succeed and move to step 2.
6) Step 2: Domain is ready to be registered, click on "register"
7) **Expect** the suggested gas amount to be much higher 
8) Again, set the gas limit low, to `26000`
9) **Expect** to see an error and back at step 2.
10) Leave the gas limit alone and proceed
11) **Expect** transaction to be successful

### CODE REVIEW

File: src/languages/en.json [hunk 1]
```diff
@@ -383,5 +383,6 @@
   "my_rsk_domain": "My RSK domain",
   "email": "Email",
   "link": "Link",
-  "whatsapp": "WhatsApp"
+  "whatsapp": "WhatsApp",
+  "transaction_receipt_failed": "Transaction failed because it did not complete. This is usually caused by not enough gas being sent."
```
Comment (ilanolkies) [inline src/languages/en.json:387]: To state _because it did not complete_ we should also check by comparing the gas used and the gas limit. I will add this response improvement to another issue.

What about adding some link here now? Like

```suggestion
  "transaction_receipt_failed": "Transaction failed. Reach us out via $Gitter$ for support."
```

File: src/app/helpers/transactionListener.js [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
-      method: 'eth_getTransactionByHash',
+      method: 'eth_getTransactionReceipt',
       params: [tx],
     }, (err, response) => {
-      if (response.result.blockNumber) {
+      if (response.result) {
         clearInterval(checkInterval);
-        dispatch(callback());
+
+        if (parseInt(response.result.status, 16) === 1) {
```
[the remark this answers is not in the document]
Comment (jessgusclark) [reply src/app/helpers/transactionListener.js:12]: Great question! The response from a successful transaction can be `1`, `true`, or `0x1`. After testing it, all three evaluate to `true`, so I'll change this to:

```
if (response.result.status) {
   return dispatch(callback());
}
```

Tested with Chrome Nifty & Metamask and Firefox Metamask. I'll test Brave this afternoon when I do a comprehensive evaluation.

Thanks.

File: src/languages/en.json (hunk 1 above)
Comment (jessgusclark) [reply src/languages/en.json:387]: Sure. The error component can't handle the link right now. Perhaps I can look at expanding the component to accept markdown and we can add it that way. For now, let's keep the string and add a link to Gitter in the footer under "developers". It should probably be there anyway.

File: src/app/helpers/transactionListener.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilanolkies) [reply src/app/helpers/transactionListener.js]: This is crazy 😆 . If we use `response.result.status` will evaluate true for `'0x0'`. The `parseInt(response.result.status, 16) === 1` was okey then, sorry.

FYI a correct implementation should follow https://github.com/ethereum/wiki/wiki/JSON-RPC#returns-31

File: src/languages/en.json (hunk 1 above)
Comment (ilanolkies) [reply src/languages/en.json]: Good! I opened https://github.com/rnsdomains/rns-manager-react/issues/301

**Pull request merged**